src/ui/public/vis/editors/default/agg.js

```diff
@@ -87,10 +83,7 @@ uiModules
 
         $scope.remove = function (agg) {
           const aggs = $scope.vis.aggs;
-
           const index = aggs.indexOf(agg);
-          if (index === -1) return notify.log('already removed');
```

Comment:
I think the index check and return should stay here.

Reply:
Great catch.